### DISCUSSION TITLE
Second I2C Bus

### DISCUSSION THREAD

**Discussion by DarthWeber** [arendst/Tasmota#19799] [Support request] [answered]
Hi, i have several I2C Sensors which work all fine with tasmota - but due to wiring i need to use a second I2C bus, which seems to be configurable in the pinning configuration (I2C_SDA-1 and -2)

But all sensors are only detected in the first bus, never on the second bus.

Do i need to configure something else?

Thanks,
Tom

**Comment by arendst** [comment]
Alas, most sensors are currently only supported on one bus.

**Comment by arendst** [comment]
Which sensors do you want to use on the second bus?

**Comment by DarthWeber** [comment]
it's the iaQ Sensor from AMS, should be number 66 or 68 in the source code.

19.10.2023 14:13:46 Theo Arends ***@***.***>:

**Comment by arendst** [comment]
Do you use an ESP8266 or an ESP32?

**Comment by arendst** [comment]
Try if this driver works 

[xsns_66_iAQ.zip](https://github.com/arendst/Tasmota/files/13043075/xsns_66_iAQ.zip)

**Comment by DarthWeber** [comment]
It's an esp32dev board, the "everything presence one" to be more specific.

Thanks, but the new code didn't fixed it, it's still the same issue: iAQ detected in I2C-1 but not on i2c-2.

**Reply by arendst** [reply, marked as answer]
In that case it's good to know it only supports one sensor in total. So on either i2c-1 OR i2c-2

Also the previous driver contained an error. This one should work on bus 2 too (when no sensor is detected on bus 1)
[xsns_66_iAQ.zip](https://github.com/arendst/Tasmota/files/13043492/xsns_66_iAQ.zip)

**Reply by DarthWeber** [reply]
Yippieh, thats works. Thanks a lot!
